source/codegen/metadata_validation.py

```diff
@@ -87,7 +87,7 @@ class RULES:
         "returns": str,
         Optional("cname"): str,
         Optional("codegen_method"): And(
-            str, lambda s: s in ("public", "private", "CustomCode", "no", "python-only")
+            str, lambda s: s in ("public", "private", "CustomCode", "no", "python-only", "CustomCodeButNoProtoMessage")
```

Comment:
Consider: would this be better as a separate tag? I think we've repeatedly run into problems where we wanted to mix-and-match `codegen_method` features.

I would also say `CustomCodeCustomProtoMessage`. You're using both "Custom" and "No" to mean "not generated but hardcoded". "Custom" is a better name for that and it's better to be consistent.

Reply:
Done.